File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Practicas/javaNapakalaki/napakalaki/PruebaNapakalaki.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;
import java.util.ArrayList;
import java.util.Arrays;




/**
 *
 * @author manjaro
 */
public final class PruebaNapakalaki{
        public static ArrayList<Monster> monstruos = new ArrayList<Monster>();

        public ArrayList<Monster> ConsultaNivel(){
           ArrayList<Monster> consultanivel = new ArrayList<Monster>();

           for(Monster m: monstruos){
               if(m.getCombatLevel() > 10){
                   consultanivel.add(m);
               }
            }
           return consultanivel;
        }

        public ArrayList<Monster> ConsultaPerdidaNivel(){
            ArrayList<Monster> perdidanivel = new ArrayList<Monster>();

            for(Monster m: monstruos){
                if((((m.getBC()).getLevels()) > 0) && (m.getBC().getnVisibleTreasures() == 0) && (m.getBC().getnHiddenTreasures() == 0) && (m.getBC().getVisibleTreasures().isEmpty()) && (m.getBC().getHiddenTreasures().isEmpty())){
                    perdidanivel.add(m);
                }
            }
            return perdidanivel;
         }



        public ArrayList<Monster> ConsultaGananciaNivel(){
            ArrayList<Monster> ganancianivel = new ArrayList<Monster>();

            for(Monster m: monstruos){
                if(m.getPremio().getLevels() > 1){
                    ganancianivel.add(m);
                }
            }
            return ganancianivel;
         }


        public ArrayList<Monster> ConsultaPerdidaTesoro(TreasureKind tesoro){
            ArrayList<Monster> perdidatesoro = new ArrayList<Monster>();

            for(Monster m: monstruos){
                if(!(m.getBC().getHiddenTreasures()).contains(tesoro) && !(m.getBC().getVisibleTreasures()).contains(tesoro)){
                } else {
                    perdidatesoro.add(m);
                }
            }
            return perdidatesoro;
         }



    /**
     * @param args the command line arguments
     */
     public void pruebaMetodos() {

        // DECLARACION DE MONSTRUOS

        // EL REY ROSADO
        BadConsequence bc1 = new BadConsequence("Pierdes 5 niveles y 3 tesoros visibles", 5, 3, 0);
        Prize p1 = new Prize(4,2);
        monstruos.add(new Monster("El rey de rosado", 13, bc1, p1));

        // ANGELES DE LA NOCHE IBICENCA || DEMONIOS DE MAGALUF
        BadConsequence bc2 = new BadConsequence("Te atrapan para llevarte de fiesta y te dejan caer en mitad del vuelo. Descarta 1 mano visible y 1 mano oculta", 0, new ArrayList<TreasureKind>(Arrays.asList(TreasureKind.ONEHAND)), new ArrayList<TreasureKind>(Arrays.asList(TreasureKind.ONEHAND)));
        Prize p2 = new Prize(4,1);
        monstruos.add(new Monster("Demonios de Magaluf", 14, bc2, p2));


        // BYANKHEES DE BONANZA

        BadConsequence bc3 = new BadConsequence("Pierdes tu armadura visible y otra oculta", 0,  new ArrayList<TreasureKind>(Arrays.asList(TreasureKind.ARMOR)),  new ArrayList<TreasureKind>(Arrays.asList(TreasureKind.ARMOR)));
        Prize p3 = new Prize(2, 1);
        monstruos.add(new Monster("Byankhees de bonanza", 8, bc3, p3));


        // TENOCHTITLAN

        BadConsequence bc4 = new BadConsequence("Embobados con el lindo primigenio te descartas de tu casco visible", 0, new ArrayList<TreasureKind>(Arrays.asList(TreasureKind.HELMET)), new ArrayList<TreasureKind>());
        Prize p4 = new Prize(1,1);
        monstruos.add(new Monster("Tenochtitlan", 2, bc4, p4));

        // EL SOPOR DE DUNWICH

        BadConsequence bc5 = new BadConsequence("El primordial bostezo contagioso. Pierdes el calzado visible", 0, new ArrayList<TreasureKind>(Arrays.asList(TreasureKind.SHOES)), new ArrayList<TreasureKind>());
        Prize p5 = new Prize(1,1);
        monstruos.add(new Monster("El sopor de Dunwich", 2, bc5, p5));

        // EL GORRON EN EL UMBRAL

        BadConsequence bc6 = new BadConsequence("Pierdes todos tus tesoros visibles", 0, 5, 0);
        Prize p6 = new Prize(3,1);
        monstruos.add(new Monster("El gorron en el umbral", 13, bc6, p6));

        // H.P. MUNCHCRAFT

        BadConsequence bc7 = new BadConsequence("Pierdes la armadura visible", 0,  new ArrayList<TreasureKind>(Arrays.asList(TreasureKind.ARMOR)), new ArrayList<TreasureKind>());
        Prize p7 = new Prize(2,1);
        monstruos.add(new Monster("H.P. Munchcraft", 6, bc7, p7));

        // NECROFAGO

        BadConsequence bc8 = new BadConsequence("Sientes bichos bajo la ropa. Descarta la armadura visible", 0, new ArrayList<TreasureKind>(Arrays.asList(TreasureKind.ARMOR)), new ArrayList<TreasureKind>());
        Prize p8 = new Prize(1,1);
        monstruos.add(new Monster("Necrofago", 13, bc8, p8));

        // FLECHER

        BadConsequence bc9 = new BadConsequence("Toses los pulmones y pierdes 2 niveles", 2, 0, 0);
        Prize p9 = new Prize(1,1);
        monstruos.add(new Monster("Flecher", 2, bc9, p9));

        // LOS HONDOS

        BadConsequence bc10 = new BadConsequence("Estos monstruos resultan bastante superficiales y te aburren mortalmente. Estas muerto", true);
        Prize p10 = new Prize(2,1);
        monstruos.add(new Monster("Los Hondos", 8, bc10, p10));

        // SEMILLAS CTHULHU

        BadConsequence bc11 = new BadConsequence("Pierdes 2 niveles y 2 tesoros ocultos", 2, 0, 2);
        Prize p11 = new Prize(2,1);
        monstruos.add(new Monster("Semillas Cthulhu", 4, bc11, p11));


        // DAMEARGO

        BadConsequence bc12 = new BadConsequence("Te intentas escaquear. Pierdes una mano visible", 0,  new ArrayList<TreasureKind>(Arrays.asList(TreasureKind.ONEHAND)), new ArrayList<TreasureKind>());
        Prize p12 = new Prize(2,1);
        monstruos.add(new Monster("Dameargo", 1, bc12, p12));

        // POLLIPOLIPO VOLANTE

        BadConsequence bc13 = new BadConsequence("Da mucho asquito. Pierdes 3 niveles", 3, 0, 0);
        Prize p13 = new Prize(2,1);
        monstruos.add(new Monster("Pollipolipo volante", 3, bc13, p13));

        // YSKHTIHYSSG-GOTH

        BadConsequence bc14 = new BadConsequence("No le hace gracia que pronuncien mal su nombre. Estas muerto", true);
        Prize p14 = new Prize(3,1);
        monstruos.add(new Monster("Yskhtihyssg-Goth", 14, bc14, p14));

        // FAMILIA FELIZ

        BadConsequence bc15 = new BadConsequence("La familia te atrapa. Estas muerto", true);
        Prize p15 = new Prize(3,1);
        monstruos.add(new Monster("Familia Feliz", 1, bc15, p15));

        // ROBOGGOTH

        BadConsequence bc16 = new BadConsequence("La quinta directiva primaria te obliga a perder 2 niveles y un tesoro 2 manos visible", 2,  new ArrayList<TreasureKind>(Arrays.asList(TreasureKind.BOTHHANDS)), new ArrayList<TreasureKind>());
        Prize p16 = new Prize(2,1);
        monstruos.add(new Monster("Roboggoth", 8, bc16, p16));

        // EL ESPIA SORDO

        BadConsequence bc17 = new BadConsequence("Te asusta en la noche. Pierdes un casco visible", 0,  new ArrayList<TreasureKind>(Arrays.asList(TreasureKind.HELMET)), new ArrayList<TreasureKind>());
        Prize p17 = new Prize(1,1);
        monstruos.add(new Monster("El espia sordo", 5, bc17, p17));

        // TONGUE

        BadConsequence bc18 = new BadConsequence("Menudo susto te llevas. Pierdes 2 niveles y 5 tesoros visibles", 2, 5, 0);
        Prize p18 = new Prize(2,1);
        monstruos.add(new Monster("Tongue", 19, bc18, p18));

        // BICEFALO

        BadConsequence bc19 = new BadConsequence("Te faltan manos para tanta cabeza. Pierdes 3 niveles y tus tesoros visibles de las manos", 3,  new ArrayList<TreasureKind>(Arrays.asList(TreasureKind.ONEHAND, TreasureKind.BOTHHANDS )), new ArrayList<TreasureKind>());
        Prize p19 = new Prize(2,1);
        monstruos.add(new Monster("Bicefalo", 21, bc19, p19));



        // ----------------------------------------------------------------------------------------------------------




        /* --------------------------------------------- Prueba Napakalaki --------------------------------------------- */








              // -------------------------------------------------------------
                // Player uno = new Player(), dos = new Player();
                //
                // BadConsequence bc = new BadConsequence("Hola", 3, 2, 2);
                // Prize p = new Prize(1,1);
                // Monster monstruo = new Monster("Monstruo-prueba", 5, bc, p);
                //
                // uno.applyPrize(monstruo);

              // -------------------------------------------------------------








/*
        // IMPRIMIR MONSTRUOS

        for(Monster m: monstruos){
           System.out.println();
           System.out.println(m.toString());
        }


        System.out.println("---------- FIN DE IMPRIMIR MONSTRUOS ----------");

        // PRUEBA DE MAIN
        System.out.println("----------------------------------------------");
        System.out.println("Prueba de consulta nivel > 10\n\n");
        PruebaNapakalaki pru = new PruebaNapakalaki();
        System.out.println((pru.ConsultaNivel()).toString());
        System.out.println("\n");

        System.out.println("----------------------------------------------");
        System.out.println("Prueba de consulta solo perdida de niveles\n\n");
        PruebaNapakalaki pru2 = new PruebaNapakalaki();
        System.out.println((pru2.ConsultaPerdidaNivel()).toString());
        System.out.println("\n");

        System.out.println("----------------------------------------------");
        System.out.println("Prueba de consulta ganancia de nivel > 1\n\n");
        PruebaNapakalaki pru3 = new PruebaNapakalaki();
        System.out.println((pru3.ConsultaGananciaNivel()).toString());
        System.out.println("\n");

        System.out.println("----------------------------------------------");
        System.out.println("Prueba de consulta de perdida de tesoro especifico\n\n");
        PruebaNapakalaki pru4 = new PruebaNapakalaki();

        System.out.println();
        System.out.println(TreasureKind.ARMOR);
        System.out.println((pru4.ConsultaPerdidaTesoro(TreasureKind.ARMOR)).toString());

        System.out.println();
        System.out.println(TreasureKind.ONEHAND);
        System.out.println((pru4.ConsultaPerdidaTesoro(TreasureKind.ONEHAND)).toString());

        System.out.println();
        System.out.println(TreasureKind.BOTHHANDS);
        System.out.println((pru4.ConsultaPerdidaTesoro(TreasureKind.BOTHHANDS)).toString());

        System.out.println();
        System.out.println(TreasureKind.HELMET);
        System.out.println((pru4.ConsultaPerdidaTesoro(TreasureKind.HELMET)).toString());

        System.out.println();
        System.out.println(TreasureKind.SHOES);
        System.out.println((pru4.ConsultaPerdidaTesoro(TreasureKind.SHOES)).toString());
        System.out.println("----------------------------------------------");
        System.out.println("\n\n");




      System.out.println("Empezamos la ejecución del programa");
      Prize pprueba = new Prize(7,8);
      BadConsequence bcprueba = new BadConsequence("BC1", 3, 2, 1);
      Monster Uno = new Monster("Monster", 20, bc1, p1);

      System.out.println(p1.getTreasures());
      System.out.println(p1.getLevels());
      System.out.println(Uno.getCombatLevel());
      System.out.println(Uno.toString());

*/
// --------------------------------------------------------------------------------------------------------
    }
}

File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Practicas/rubyNapakalaki/pruebaNapakalaki.rb
#encoding: utf-8
=begin
  Para permitir uso de tildes
=end
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "treasure.rb"
require_relative "monster.rb"
require_relative "treasure_kind.rb"
require_relative "combat_result.rb"
require_relative "dice.rb"
require_relative "prize.rb"
require_relative "card_dealer.rb"
require_relative "bad_consequence.rb"
require_relative "napakalaki.rb"
require_relative "player.rb"

module Napakalaki


class PruebaNapakalaki

=begin


  def self.main #Definir main como metodo para luego llamarlo

    @@monsters = Array.new


    # EL REY DE ROSADO

    p1 = Prize.new(4,2)
    bC1 = BadConsequence.newLevelNumberOfTreasures("Pierdes 5 niveles y 3 tesoros visibles", 5, 3, 0)
    @@monsters << Monster.new("El rey de rosado", 13, p1, bC1)


    # DEMONIOS DE MAGALUF

    p2 = Prize.new(4,1)
    bC2 = BadConsequence.newLevelSpecificTreasures("Te atrapan para llevarte de fiesta y te dejan caer en mitad del vuelo. Descarta 1 mano visible y 1 oculta", 0, [TreasureKind::ONEHAND], [TreasureKind::ONEHAND])
    @@monsters << Monster.new("Demonios de Magaluf", 14, p2, bC2)


    # BYANKHEES DE BONANZA

    p3 = Prize.new(2,1)
    bC3 = BadConsequence.newLevelSpecificTreasures("Pierdes tu armadura visible y otra oculta", 0, [TreasureKind::ARMOR], [TreasureKind::ARMOR])
    @@monsters << Monster.new("Byankhees de bonanza", 8, p3, bC3)


    # TENOCHTITLAN

    p4 = Prize.new(1,1)
    bC4 = BadConsequence.newLevelSpecificTreasures("Embobados con el lindo primigenio te descartas de tu casco visible", 0, [TreasureKind::HELMET], nil)
    @@monsters << Monster.new("Tenochtitlan", 2, p4, bC4)


    # EL SOPOR DE DUNWICH

    p5 = Prize.new(1,1)
    bC5 = BadConsequence.newLevelSpecificTreasures("El primordial bostezo contagioso. Pierdes el calzado visible", 0, [TreasureKind::SHOES], nil)
    @@monsters << Monster.new("El sopor de Dunwich", 2, p5, bC5)


    # EL GORRON EN EL UMBRAL

    p6 = Prize.new(3,1)
    bC6 = BadConsequence.newLevelNumberOfTreasures("Pierdes todos tus tesoros visibles", 0, 5, 0)
    @@monsters << Monster.new("El gorron en el umbral", 13, p6, bC6)


    # H.P. MUNCHCRAFT

    p7 = Prize.new(2,1)
    bC7 = BadConsequence.newLevelSpecificTreasures("Pierdes la armaudura visible", 0, [TreasureKind::ARMOR], nil)
    @@monsters << Monster.new("H.P. Munchcraft", 6, p7, bC7)


    # NECROFAGO

    p8 = Prize.new(1,1)
    bC8 = BadConsequence.newLevelSpecificTreasures("Sientes bichos bajo la ropa. Descarta la armadura visible", 0, [TreasureKind::ARMOR], nil)
    @@monsters << Monster.new("Necrofago", 13, p8, bC8)


    # FLECHER

    p9 = Prize.new(1,1)
    bC9 = BadConsequence.newLevelNumberOfTreasures("Toses los pulmones y pierdes 2 niveles", 2, 0, 0)
    @@monsters << Monster.new("Flecher", 2, p9, bC9)


    # LOS HONDOS

    p10 = Prize.new(2,1)
    bC10 = BadConsequence.newDeath("Estos monstruos resultan bastante superficiales y te aburren mortalmente. Estas muerto")
    @@monsters << Monster.new("Los Hondos", 8, p10, bC10)

    # SEMILLAS CTHULHU

    p11 = Prize.new(2,1)
    bC11 = BadConsequence.newLevelNumberOfTreasures("Pierdes 2 niveles y 2 tesoros ocultos", 2, 0, 2)
    @@monsters << Monster.new("Semillas Cthulhu", 4, p11, bC11)


    # DAMEARGO

    p12 = Prize.new(2,1)
    bC12 = BadConsequence.newLevelSpecificTreasures("Te intentas escaquear. Pierdes una mano visible", 0, [TreasureKind::ONEHAND], nil)
    @@monsters << Monster.new("Dameargo", 1, p12, bC12)


    # POLLIPOLIPO VOLANTE

    p13 = Prize.new(2,1)
    bC13 = BadConsequence.newLevelNumberOfTreasures("Da mucho asquito. Pierdes 3 niveles", 3, 0, 0)
    @@monsters << Monster.new("Pollipolipo volante", 3, p13, bC13)


    # YSKHTIHYSSG-GOTH

    p14 = Prize.new(3,1)
    bC14 = BadConsequence.newDeath("No le hace gracia que pronuncien mal su nombre. Estas muerto")
    @@monsters << Monster.new("Yskhtihyssg-Goth", 14, p14, bC14)


    # FAMILIA FELIZ

    p15 = Prize.new(3,1)
    bC15 = BadConsequence.newDeath("La familia te atrapa. Estas muerto")
    @@monsters << Monster.new("Familia feliz", 1, p15, bC15)


    # ROBOGGOTH

    p16 = Prize.new(2,1)
    bC16 = BadConsequence.newLevelSpecificTreasures("La quinta directiva primaria te obliga a perder 2 niveles y un tesoro 2 manos visible", 2, [TreasureKind::BOTHHANDS], nil)
    @@monsters << Monster.new("Roboggoth", 8, p16, bC16)


    # EL ESPIA SORDO

    p17 = Prize.new(1,1)
    bC17 = BadConsequence.newLevelSpecificTreasures("Te asusta en la noche. Pierdes un casco visible", 0, [TreasureKind::HELMET], nil)
    @@monsters << Monster.new("El espia sordo", 5, p17, bC17)


    # TONGUE

    p18 = Prize.new(2,1)
    bC18 = BadConsequence.newLevelNumberOfTreasures("Menudo susto te llevas. Pierdes 2 niveles y 5 tesoros visibles", 2, 5, 0)
    @@monsters << Monster.new("Tongue", 19, p18, bC18)


    # BICEFALO

    p19 = Prize.new(2,1)
    bC19 = BadConsequence.newLevelSpecificTreasures("Te faltan manos para tanta cabeza. Pierdes 3 niveles y tus tesoros visibles de las manos", 3, [TreasureKind::ONEHAND, TreasureKind::BOTHHANDS], nil)
    @@monsters << Monster.new("Bicefalo", 21, p19, bC19)



    # NIVEL COMBATE > 10

    def self.combatsup10
      clvl = Array.new
      @@monsters.each do |m|
        if m.combatLevel > 10
          clvl << m
        end
      end

      for i in 0...clvl.length
         puts clvl[i].to_s
      end
    end


    # BADCONSEQUENCE SOLO PERDIDA DE NIVEL

    def self.perdidanivel
      plvl = Array.new
      @@monsters.each do |m|
        if m.bC.levels > 0 && m.bC.nVisibleTreasures == 0 && m.bC.nHiddenTreasures == 0 && m.bC.specificVisibleTreasures == nil && m.bC.specificHiddenTreasures == nil
          plvl << m
        end
      end

      for i in 0...plvl.length
         puts plvl[i].to_s
      end
    end


    # PRIZE CON GANANCIA DE NIVEL SUPERIOR A 1

     def self.ganancianivel
      glvl = Array.new
      @@monsters.each do |m|
        if m.prize.level > 1
          glvl << m
        end
      end

      for i in 0...glvl.length
         puts glvl[i].to_s
      end
    end

    # PERDIDA DE TESORO ESPECIFICO

    def self.perdidatesoro(uno)
      res = Array.new
      @@monsters.each do |m|
        if m.bC.specificVisibleTreasures == uno || m.bC.specificHiddenTreasures == uno
          res << m
        end
      end

      for i in 0...res.length
         puts res[i].to_s
      end
    end



#MAIN










# MAIN
    puts "----------------------------------------------------\n\n"
    puts "Monstruos con nivel de combate superior a 10" + "\n\n"
    PruebaNapakalaki.combatsup10
    puts "----------------------------------------------------\n\n"

    puts "BadConsequence solo perdida de nivel" + "\n\n"
    PruebaNapakalaki.perdidanivel
    puts "----------------------------------------------------\n\n"

    puts "Ganancia de nivel superior a 1" + "\n\n"
    PruebaNapakalaki.ganancianivel
    puts "----------------------------------------------------\n\n"

    puts "Perdida de un determinado tipo de tesoro" + "\n\n"
    puts TreasureKind::ARMOR
    puts"\n"
    PruebaNapakalaki.perdidatesoro([TreasureKind::ARMOR])
    puts "\n\n"
    puts TreasureKind::ONEHAND
    puts"\n"
    PruebaNapakalaki.perdidatesoro([TreasureKind::ONEHAND])
    puts "\n\n"
    puts TreasureKind::BOTHHANDS
    puts"\n"
    PruebaNapakalaki.perdidatesoro([TreasureKind::BOTHHANDS])
    puts "\n\n"
    puts TreasureKind::HELMET
    puts"\n"
    PruebaNapakalaki.perdidatesoro([TreasureKind::HELMET])
    puts "\n\n"
    puts TreasureKind::SHOES
    puts"\n"
    PruebaNapakalaki.perdidatesoro([TreasureKind::SHOES])
    puts"\n\n"
    puts "----------------------------------------------------\n\n"

# -------------------------------------------------------------------------------
end

PruebaNapakalaki.main #llamar a main

=end

end
end
